src/test_util/mock/transforms/mod.rs

```diff
@@ -1,8 +1,22 @@
-use crate::config::TransformDescription;
+use vector_config::configurable_component;
 
 mod basic;
 pub use self::basic::BasicTransformConfig;
 
-inventory::submit! {
-    TransformDescription::new::<BasicTransformConfig>("test_basic")
+mod noop;
+pub use self::noop::NoopTransformConfig;
+
+/// Transform types.
+#[configurable_component]
+#[derive(Clone, Debug)]
+pub enum TransformType {
```

Comment:
Definitely a comment for later work/TODOs - but these kinds feel very disjointed name wise. It might be nice cleanup to find names that are more clear of the intention at first glance. ex: `Task` => `Async` or similar if it's differentiated by async processing.

Reply:
Yeah, I also struggled when trying to write the description for them.

Reply:
Definitely not an issue for this PR, they've been around for a while 👍 don't really have great ideas either for renaming